packages/cli/src/processed-data-managers/processed-data-helper.ts

```diff
@@ -0,0 +1,322 @@
+					returnData.new.push(item);
+					if (ProcessedDataHelper.compareValues(options.mode, item, largestValue)) {
+						largestValue = item;
+					}
+				} else {
+					returnData.processed.push(item);
+				}
+			});
+
+			processedData.value.data = largestValue;
+
+			await Container.get(ProcessedDataRepository).save(processedData);
+
+			return returnData;
+		}
+
+		const hashedItems = items.map((item) => ProcessedDataHelper.createValueHash(item));
+
+		if (!processedData) {
+			// All items are new so add new entries
+			if (options.maxEntries) {
+				hashedItems.splice(0, hashedItems.length - options.maxEntries);
+			}
+			await Container.get(ProcessedDataRepository).insert({
+				workflowId: contextData.workflow.id.toString(),
+				context: dbContext,
+				value: {
+					mode: options.mode,
+					data: hashedItems,
+				},
+			});
+
+			return {
+				new: items,
+				processed: [],
+			};
+		}
+
+		const returnData: ICheckProcessedOutput = {
+			new: [],
+			processed: [],
+		};
+
+		const processedDataValue = processedData.value as IProcessedDataEntries;
+
+		hashedItems.forEach((item, index) => {
+			if (processedDataValue.data.find((entry) => entry === item)) {
+				returnData.processed.push(items[index]);
+			} else {
+				returnData.new.push(items[index]);
+				processedDataValue.data.push(item);
+			}
+		});
+
+		if (options.maxEntries) {
+			processedDataValue.data.splice(0, processedDataValue.data.length - options.maxEntries);
+		}
+
+		await Container.get(ProcessedDataRepository).save(processedData);
+
+		return returnData;
+	}
+
+	async removeProcessed(
+		items: ProcessedDataItemTypes[],
+		context: ProcessedDataContext,
+		contextData: ICheckProcessedContextData,
+		options: ICheckProcessedOptions,
+	): Promise<void> {
+		if (['latestIncrementalKey', 'latestDate'].includes(options.mode)) {
+			throw new ApplicationError('Removing processed data is not possible in mode "latest"');
+		}
+
+		const processedData = await Container.get(ProcessedDataRepository).findOne({
+			where: {
+				workflowId: contextData.workflow.id as string,
+				context: ProcessedDataHelper.createContext(context, contextData),
+			},
+		});
+
+		if (!processedData) {
+			return;
+		}
+
+		const hashedItems = items.map((item) => ProcessedDataHelper.createValueHash(item));
+
+		const processedDataValue = processedData.value as IProcessedDataEntries;
+
+		hashedItems.forEach((item) => {
+			const index = processedDataValue.data.findIndex((value) => value === item);
+			if (index !== -1) {
+				processedDataValue.data.splice(index, 1);
+			}
+		});
+
+		await Container.get(ProcessedDataRepository).save(processedData);
+	}
+
+	async clearAllProcessedItems(
+		context: ProcessedDataContext,
+		contextData: ICheckProcessedContextData,
+		options: ICheckProcessedOptions,
+	): Promise<void> {
+		console.log(options);
+		await Container.get(ProcessedDataRepository).delete({
+			workflowId: contextData.workflow.id as string,
```

Comment:
Let's assert that this exists like we do in the other method so there's no need for a cast